Topics/Tech_Stacks/EJS.md

```diff
@@ -0,0 +1,97 @@
+  
+- **Conditional Rendering**: Implement conditional logic to selectively render content based on specified criteria
+
+- **Data Binding**: Bind server-side data to client-side views, facilitating the presentation of dynamic information to users.
+
+- **Iterative Rendering**: Iterate over data collections to dynamically generate HTML elements, enabling the creation of dynamic lists, tables, and more.
+
+## Installation
+
+Before using EJS, you need to install it. You can install EJS via npm, the Node.js package manager, using the following command:
+
+```bash
+npm install ejs
+```
+
+## Getting Started
+
+To begin using EJS, create a new HTML file and import the EJS library:
+```html
+<script src="ejs.min.js"></script>
+```
+
+## Basic Syntax and Usage
+EJS employs a very simple but powerful syntax to embed JavaScript code within HTML markup. Below are some of the fundamental concepts:
+
+- **Interpolation: Use <%= %> tags to insert the result of a JavaScript expression into the HTML document.
+
+- **Conditional Statements: Use <% if (condition) { %> ... <% } %> blocks to implement conditional rendering based on specific conditions.
+
+- **Iterative Rendering: Use <% for (let item of items) { %> ... <% } %> constructs to iterate over data collections and generate dynamic content.
+
+## Advanced Features
+Beyond the basics, EJS offers advanced features to further enhance your web development workflow:
+
+- **Partials: Convert your templates into modules by creating reusable partials which enables you to reuse your code and make it look more presentable, similar to React components
+
+- **Layouts: Define master layouts to encapsulate common page structures and streamline the creation of consistent user interfaces.
+
+- **Custom Functions: Extend EJS functionality by defining custom JavaScript functions and incorporating them into your templates.
+
+## Example Usage
+In this section we see how some of these functionalities are used
+
+This is an example of a partial:
+```html
+</div>
+<div class="footer-padding"></div>
+  <div class="footer">
+    <p>Copyright</p>
+  </div>
+</div>
+</body>
+</html>
+```
+
+This footer can be reused in any other pages by including this line whereever the footer is needed. 
+
+```html
+<%- include("partials/header") %>
+
+<h1><%= postHeader %></h1>
+<p> <%= postContent %> </p>
+
+<%- include("partials/footer") %>
+```
+
+In this example, we include the header, have some custom content, and then have the footer. This also shows how interpolation is used to display the variables postHeader and postContent which are passed in as JSON objects by the server. 
+
+Below we see an example of iterative rendering:
+```html
+<% posts.forEach(function(post) { %>
+  <h1> <%= post.title %></h1>
+  <p>
+    <%= post.content.substring(0, 100) + " ..."%>
+    <a href="/posts/ <%=post.title %>">Read More</a>
+  </p>
+<% }) %>
+```
+
+
```

Comment:
You could give more general examples or a layout of how implementation might look for embedded JS. I feel like more examples of such would better help CSC301 students get the best out of your page and be able to better apply their learning.